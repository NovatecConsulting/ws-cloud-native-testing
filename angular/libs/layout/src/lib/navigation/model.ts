import { Type } from '@angular/core';

export class DefinedRoute {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  constructor(public path: string, public component: Type<any>, public displayName: string) {}

  buildRouterLink() {
    return '/' + this.path.replace('**', '');
  }
}
