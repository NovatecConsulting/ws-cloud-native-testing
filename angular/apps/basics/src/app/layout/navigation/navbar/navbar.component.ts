import { Component } from '@angular/core';
import { DEFINED_ROUTES } from '../../../routes';

@Component({
  selector: 'basics-navbar',
  templateUrl: './navbar.component.html',
  styleUrls: ['./navbar.component.scss'],
})
export class NavbarComponent {
  routes = DEFINED_ROUTES.filter((route) => !!route.displayName);
}
