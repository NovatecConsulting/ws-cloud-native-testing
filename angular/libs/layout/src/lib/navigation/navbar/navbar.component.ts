import { Component, Input } from '@angular/core';
import { DefinedRoute } from '../model';

@Component({
  selector: 'layout-navbar',
  templateUrl: './navbar.component.html',
  styleUrls: ['./navbar.component.scss'],
})
export class NavbarComponent {
  @Input() routes: DefinedRoute[] = [];
}
