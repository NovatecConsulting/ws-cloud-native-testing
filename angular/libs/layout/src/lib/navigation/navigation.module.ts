import { NgModule } from '@angular/core';
import { MatToolbarModule } from '@angular/material/toolbar';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { MatButtonModule } from '@angular/material/button';
import { NavbarComponent } from './navbar/navbar.component';

@NgModule({
  declarations: [NavbarComponent],
  imports: [CommonModule, RouterModule, MatToolbarModule, MatButtonModule],
  exports: [NavbarComponent],
})
export class NavigationModule {}
