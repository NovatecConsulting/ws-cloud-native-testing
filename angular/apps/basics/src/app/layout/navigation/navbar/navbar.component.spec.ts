import { MockBuilder, MockRender } from 'ng-mocks';
import { NavbarComponent } from './navbar.component';
import { NavigationModule } from '../navigation.module';
import { RouterTestingModule } from '@angular/router/testing';

describe('Navbar', () => {
  beforeEach(() => MockBuilder(NavbarComponent, NavigationModule).keep(RouterTestingModule));

  it('should stay consistent', () => {
    const fixture = MockRender(NavbarComponent);
    expect(fixture).toMatchSnapshot();
  });
});
