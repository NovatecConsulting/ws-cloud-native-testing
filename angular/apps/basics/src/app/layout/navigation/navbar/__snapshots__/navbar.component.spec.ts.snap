// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should stay consistent 1`] = `
<mock-render
  __ngContext__={[Function LRootView]}
>
  <basics-navbar>
    <mat-toolbar
      class="w-screen"
      color="primary"
    >
      <a
        href="/weather"
        mat-button=""
      >
         Weather 
      </a>
      <a
        href="/"
        mat-button=""
      >
         Home 
      </a>
    </mat-toolbar>
  </basics-navbar>
</mock-render>
`;
